src/genome_builds.php

```diff
@@ -124,46 +141,69 @@
                     'position_g_start_' . $_POST['column_suffix'],
                     'position_g_end_' . $_POST['column_suffix']
                 ))->fetchAllColumn();
-            if (count($aActiveColumns) < 3) {
-                $sSQL = 'ALTER TABLE ' . TABLE_VARIANTS;
-                if (!in_array('VariantOnGenome/DNA/' . $_POST['column_suffix'], $aActiveColumns)) {
-                    $sSQL .= ' ADD COLUMN `VariantOnGenome/DNA/' . $_POST['column_suffix'] . '` VARCHAR(255),';
-                }
-                if (!in_array('position_g_start_' . $_POST['column_suffix'], $aActiveColumns)) {
-                    $sSQL .= ' ADD COLUMN position_g_start_' . $_POST['column_suffix'] . ' INT(10) UNSIGNED AFTER position_g_end,';
-                }
-                if (!in_array('position_g_end_' . $_POST['column_suffix'], $aActiveColumns)) {
-                    $sSQL .= ' ADD COLUMN position_g_end_' . $_POST['column_suffix'] . ' INT(10) UNSIGNED AFTER position_g_start_' . $_POST['column_suffix'] . ',';
+            $aActiveColumnsTranscripts = $_DB->query('
+                SELECT COLUMN_NAME
+                FROM information_schema.COLUMNS
+                WHERE TABLE_SCHEMA = DATABASE()
+                  AND TABLE_NAME = "' . TABLE_TRANSCRIPTS . '"
+                  AND COLUMN_NAME IN (?,?)',
+                array(
+                    'position_g_mrna_start_' . $_POST['column_suffix'],
+                    'position_g_mrna_end_' . $_POST['column_suffix']))->fetchAllColumn();
+
+            $aTablesAndTheirColumns = array(
+                TABLE_VARIANTS => array(
+                    '`VariantOnGenome/DNA/' . $_POST['column_suffix'] . '`' => array(
+                        'type' => 'VARCHAR(255)',
+                        'active' => in_array(
+                            'VariantOnGenome/DNA/' . $_POST['column_suffix'],
+                            $aActiveColumnsVariants)
+                    ),
```

Comment:
This still has a lot of repetition, and it's still quite long. I will update this to show a much shorter alternative.